core/common/src/main/java/alluxio/grpc/GrpcConnection.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.grpc;
+
+import alluxio.conf.AlluxioConfiguration;
+
+import io.grpc.Channel;
+import io.grpc.ClientInterceptor;
+import io.grpc.ClientInterceptors;
+import io.grpc.ManagedChannel;
+
+import java.util.Objects;
+
+/**
+ * Used to gather gRPC level resources and indexes together.
+ */
+public class GrpcConnection implements AutoCloseable {
+
+  private GrpcConnectionKey mKey;
+  private ManagedChannel mManagedChannel;
+  private Channel mChannel;
```

Comment:
These two are always the same?

Reply:
`ManagedChannel` is the original netty side channel.
`Channel` is usually that channel wrapped with different interceptors. For example; each logical channel need to inject its own channel-id so they have their own interceptors. So we keep wrapping this channel until it has the final logical state we want.